projects/slo-libsonnet.yml

```diff
@@ -0,0 +1,5 @@
+name: slo-libsonnet
```

Comment:
Not sure if this one belongs in this list ...

The purpose of this repo is to collect Kubernetes specific libraries:
> * Modules provide Kubernetes API resources to install entire Applications. They can be compared to Helm Charts.
> * Mixins usually provide only configuration of some kind. This can be for example Grafana dashboards, Prometheus alerts, etc

`slo-libsonnet`, like `grafonnet` or `ksonnet-lib` are helper libraries that can be used to create modules or mixins, but are not directly usable on their own.

It would of course make sense to gather such helpers in a central place as well, maybe as a new category in this project, or instead on the more general  [metalmatze/awesome-libsonnet](https://github.com/metalmatze/awesome-libsonnet). /cc @metalmatze

Reply:
> The purpose of this repo is to collect Kubernetes specific libraries:

That statement confuses me a bit. Mixins literally are about _not_ being Kubernetes specific, obviously with the exception of the kubernetes-mixin. I think I would agree that it might make sense to have a more general collection and not make it kubernetes focused. Happy to have this join forces with @metalmatze's repo, I don't have a strong opinion about which would merge into which one.

Reply:
Okay, out of context this sounds indeed confusing.

The rationale behind this repository is to collect:

1. Libraries that create Kubernetes manifests for a certain application (**Modules**)
2. Libraries that allow to monitor above modules using precomposed dashboards, rules, alerts, etc (**Mixins**).

For a user, this should result in installing an Application and having it properly monitored ootb.

`slo-libsonnet`, `grafonnet` et al do not exactly fit these criteria, because they do neither create k8s resources, nor dashboards, etc. Instead, they are helper tools to create modules and mixins. This of course does not mean they should not be gathered somewhere (maybe even in a new section in this repo's README).

Is this more clear?

Reply:
So something like a catalog? I think that's a cool idea (in fact we started some design docs around full packages for this type of stuff internally), but I think the name of the project implies something else, as the `awesome-*` kind of projects are ussually an arbitrary collection of things. I would suggest renaming the project to something that reflects its purpose a bit better.

Reply:
Sounds fair, this repo is temporary after all.

In the long run, I'd prefer something much more sophisticated (e.g. `pkg.tanka.dev` that hosts like hub.helm.sh modules, matching mixins, jsonnet docs for those, etc)

Reply:
This one is really about gathering structured data about existing libraries to be built into something bigger soon

Reply:
Agreed and as mixins are intended to be usable outside of the kube context I think we should have that even be an entirely separate thing and then ensure that the mechanisms built for the kube context harmonize with it.